x/slashing/params.go

```diff
@@ -4,77 +4,114 @@ import (
 	"time"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/params"
 )
 
-// nolint
+// Default parameter namespace
 const (
-	MaxEvidenceAgeKey           = "slashing/MaxEvidenceAge"
-	SignedBlocksWindowKey       = "slashing/SignedBlocksWindow"
-	MinSignedPerWindowKey       = "slashing/MinSignedPerWindow"
-	DoubleSignUnbondDurationKey = "slashing/DoubleSignUnbondDuration"
-	DowntimeUnbondDurationKey   = "slashing/DowntimeUnbondDuration"
-	SlashFractionDoubleSignKey  = "slashing/SlashFractionDoubleSign"
-	SlashFractionDowntimeKey    = "slashing/SlashFractionDowntime"
+	DefaultParamspace = "slashing"
 )
 
+// Parameter store key
+var (
+	KeyMaxEvidenceAge           = []byte("MaxEvidenceAge")
+	KeySignedBlocksWindow       = []byte("SignedBlocksWindow")
+	KeyMinSignedPerWindow       = []byte("MinSignedPerWindow")
+	KeyDoubleSignUnbondDuration = []byte("DoubleSignUnbondDuration")
+	KeyDowntimeUnbondDuration   = []byte("DowntimeUnbondDuration")
+	KeySlashFractionDoubleSign  = []byte("SlashFractionDoubleSign")
+	KeySlashFractionDowntime    = []byte("SlashFractionDowntime")
+)
+
+// Params - used for initializing default parameter for slashing at genesis
+type Params struct {
+	MaxEvidenceAge           time.Duration `json:"max-evidence-age"`
+	SignedBlocksWindow       int64         `json:"signed-blocks-window"`
+	MinSignedPerWindow       sdk.Dec       `json:"min-signed-per-window"`
+	DoubleSignUnbondDuration time.Duration `json:"doublesign-unbond-duration"`
+	DowntimeUnbondDuration   time.Duration `json:"downtime-unbond-duration"`
+	SlashFractionDoubleSign  sdk.Dec       `json:"slash-fraction-doublesign"`
+	SlashFractionDowntime    sdk.Dec       `json:"slash-fraction-downtime"`
+}
+
+// Implements params.ParamStruct
+func (p *Params) KeyFieldPairs() params.KeyFieldPairs {
+	return params.KeyFieldPairs{
+		{KeyMaxEvidenceAge, &p.MaxEvidenceAge},
+		{KeySignedBlocksWindow, &p.SignedBlocksWindow},
+		{KeyMinSignedPerWindow, &p.MinSignedPerWindow},
+		{KeyDoubleSignUnbondDuration, &p.DoubleSignUnbondDuration},
+		{KeyDowntimeUnbondDuration, &p.DowntimeUnbondDuration},
+		{KeySlashFractionDoubleSign, &p.SlashFractionDoubleSign},
+		{KeySlashFractionDowntime, &p.SlashFractionDowntime},
+	}
+}
+
+// Default parameters used by Cosmos Hub
+func DefaultParams() Params {
+	return Params{
+		// defaultMaxEvidenceAge = 60 * 60 * 24 * 7 * 3
+		// TODO Temporarily set to 2 minutes for testnets.
```

Comment:
I'd actually prefer we left these as what we want the defaults to be, and we modified them within the gaia package. Can be saved for a separate PR though.

Reply:
How about defining a new func `TestnetParams` and separate two cases? I will open an issue for it.

Reply:
These should really be in genesis too